Plutarch/Api/V1/AssocMap.hs

```diff
@@ -1,14 +1,49 @@
+{-# LANGUAGE TemplateHaskell #-}
```

Comment:
Where are we using template haskell?

Reply:
In order to generate the instances for `MapUnion`, so I could use `letrec`.